### PR TITLE
Add "GSoC 2024: ABI Lowering in ClangIR" blog post

### PR DESCRIPTION
**Pull request by sitio-couto**
CC: @asl @bcardosolopes @lanza

### CODE REVIEW

File: content/posts/2024-09-07-abi-lowering-in-clangir.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+- Improving DataLayout-related queries using MLIR's built-in tools.
+- Implementing calling convention lowering for additional types, such as
+  pointers.
+- Extending the TargetLowering library to support more architectures.
+- Unifying remaining target-specific lowering code from other parts of ClangIR.
+
+## Acknowledgements
+
+I would like to thank my Google Summer of Code mentors, Bruno Cardoso and Nathan
```
Comment (bcardosolopes) [inline content/posts/2024-09-07-abi-lowering-in-clangir.md:94]: Bruno Cardoso Lopes

File: content/posts/2024-09-07-abi-lowering-in-clangir.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+---
+author: "Vinicius Espindola"
+date: "2024-09-07"
+tags: ["GSoC", "mlir", "clangir"]
+title: "GSoC 2024: ABI Lowering in ClangIR"
+---
+
+ClangIR is an ongoing effort to build a high-level intermediate representation
+(IR) for C++ within the LLVM ecosystem. Its key advantage lies in its ability to
```
Comment (bcardosolopes) [inline content/posts/2024-09-07-abi-lowering-in-clangir.md:9]: *C/C++

**Review by bcardosolopes — APPROVED** [review]
LGTM after some minor fixes

**Comment by sitio-couto** [comment]
@bcardosolopes applied

**Review by bcardosolopes — APPROVED** [review]

File: content/posts/2024-09-07-abi-lowering-in-clangir.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+---
+author: "Vinicius Espindola"
+date: "2024-09-07"
```
Comment (asl) [inline content/posts/2024-09-07-abi-lowering-in-clangir.md:3]:
```suggestion
date: "2024-09-30"
```

**Review by asl — APPROVED** [review]
Scheduled for September 30

**Pull request merged**